### DISCUSSION TITLE
Allow reordering records in the List view via drag-and-drop

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#1413] [Roadmap]
We should support reordering document in the List view by some specified field property. It should support drag-and-drop.

**Comment by coxmi** [comment]
Would you also consider a hierarchical list view, with drag and drop?

**Reply by jmikrut** [reply]
Yes, but I think this would be a separate feature request. Still could be doable!

**Comment by quornik** [comment]
Reordering bu also delete and bulk delete of records, would be great features.

**Reply by jmikrut** [reply]
Yeah! That is also on our radar and ..... prioritized first!

https://github.com/payloadcms/payload/discussions/1382

**Comment by jacobsfletch** [comment]
Related: https://github.com/payloadcms/payload/discussions/1498

**Comment by phillipmalboeuf** [comment]
Hey gang @jmikrut @jacobsfletch, I've started work on this, see discord thread: [thread](https://discord.com/channels/967097582721572934/1063111110070370324) and [github compare](https://github.com/payloadcms/payload/compare/master...phillipmalboeuf:payload:feat/sortable-columns)
Could you let me know what kind of tests and documentation would be needed for it?

**Reply by JarrodMFlesch** [reply]
This is very cool.

**Comment by phillipmalboeuf** [comment]
Opened a PR: https://github.com/payloadcms/payload/pull/1865

**Comment by 58bits** [comment]
Hi all - have not looked at this closely, nor at @phillipmalboeuf 's solution  - but thought you might find this interesting  https://youtu.be/5SQkOyzZLHM  (although we're also a @dnd-kit user)

**Comment by r1tsuu** [comment]
I was looking for another discussion, but i have a plugin that does exactly that https://github.com/r1tsuu/payload-plugin-collections-docs-order (i believe the PR that merged here is about "column sorting in the list vew", not "document sorting by a specific field".
Any chance it could be in the core of Payload / or as a plugin in the Payload repo? If yes, i can help with a PR, just give me some feedback about how well or not i done that and maybe suggestions to the code usage/uiux.

**Reply by architechx** [reply]
are you no longer maintaining this? I tried installing but ran into a few problems

**Comment by 58bits** [comment]
@r1tsuu - thanks for this - this is great. We'd also love to see this feature in core - ideally as an option of the main list view - although your solution is also excellent . Our current approach is to create an integer-based 'order' column, which we manually place padded index / order values in - which isn't ideal. Below is a screen shot for a 'Bios' listing, which we regularly use in pages like 'Our Team' etc.
![sorted](https://github.com/payloadcms/payload/assets/405612/c79b9864-488f-41a4-bfb0-5565b775c333)
Will take a closer look at your plugin soon ;-)

**Comment by r1tsuu** [comment]
> @r1tsuu - thanks for this - this is great. We'd also love to see this feature in core - ideally as an option of the main list view - although your solution is also excellent . Our current approach is to create an integer-based 'order' column, which we manually place padded index / order values in - which isn't ideal. Below is a screen shot for a 'Bios' listing, which we regularly use in pages like 'Our Team' etc. ![sorted](https://private-user-images.githubusercontent.com/405612/320232341-c79b9864-488f-41a4-bfb0-5565b775c333.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI0MzkzMjAsIm5iZiI6MTcxMjQzOTAyMCwicGF0aCI6Ii80MDU2MTIvMzIwMjMyMzQxLWM3OWI5ODY0LTQ4OGYtNDFhNC1iZmIwLTU1NjViNzc1YzMzMy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwNlQyMTMwMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YmMxNDgzZmQxNzUwZDJmZjdkZWJlYWJkMzY4MTJiMWJlNWE4MjhiN2U2MWE5NjcwODdiNDAxNjgzOGY2OTQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.dSgotzfukU78BEqZP0zXaLE6nv8kq12lQyvU1gUvOCY) Will take a closer look at your plugin soon ;-)

We also used this approach, but i believe it was looking kinda bad for clients. My plugin now solves this for me quite well, if you'll find something that could be improved or added you can freely create an issue / PR

**Comment by bespoke-dungnguyen** [comment]
I want this feature

**Comment by LuVe96** [comment]
In addition to reordering, it may be worthwhile to structure nested documents (via the nestedDocsPlugin) in a tree view, allowing for drag-and-drop customization.

For example, the CMS [Statamic](https://statamic.com/) offers a similar feature. Alongside the default list view, it includes a tree view where entries can be reordered via drag-and-drop. The tree view also supports nesting, meaning entries with a parent are displayed hierarchically beneath their parent entry.

![Statamic Tree View](https://github.com/user-attachments/assets/b52848c3-c85c-428f-b21b-328a7ac588ef)

**Comment by jeremybasolo** [comment]
up, it would be great to have this feature into core

**Comment by GermanJablo** [comment]
[This feature is already done](https://github.com/payloadcms/payload/pull/11452)!

In the next release, there will be two major improvements:
1. You will be able to add `orderable: true` to a collection with existing documents (currently, you'll get an error).
2. [The DnD experience will be improved](https://github.com/payloadcms/payload/pull/11959), thanks to @akhrarovsaid!

Thank you all for your comments!

**Reply by claster-studio** [reply]
thx from 2025

**Comment by joshdavenport** [comment]
That's incredible! Thanks so much for the hard work everyone

**Comment by kristian240** [comment]
Hi @GermanJablo @jmikrut 👋🏼 ! There are some issues with this feature, some of it are already done (https://github.com/payloadcms/payload/commit/e90ff72b3742c34d10745b9349a1359a8ecca826 and https://github.com/payloadcms/payload/commit/3287f7062f3e1d1f06628e0c21e77addec752bf8) but not released yet and some have open PRs (https://github.com/payloadcms/payload/pull/12129). When can we get new changes released?
